# treat life_expectency data base

In [7]:
import pandas as pd

In [8]:
data = pd.read_csv('/Users/emilytsen/Documents/projects/ds_cancer/df/life_expectancy.csv')

In [9]:
print(data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3253 entries, 0 to 3252
Data columns (total 3 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Entity           3253 non-null   object 
 1   Year             3253 non-null   int64  
 2   Life expectancy  3253 non-null   float64
dtypes: float64(1), int64(1), object(1)
memory usage: 76.4+ KB
None


In [10]:
print(data.head())

      Entity  Year  Life expectancy
0  Australia  1802        34.049999
1  Australia  1803        34.049999
2  Australia  1804        34.049999
3  Australia  1805        34.049999
4  Australia  1806        34.049999


In [11]:
print(data.describe())

              Year  Life expectancy
count  3253.000000      3253.000000
mean   1908.066093        48.680380
std      62.613962        17.965669
min    1800.000000         8.108836
25%    1854.000000        32.000000
50%    1908.000000        41.880001
75%    1962.000000        66.820000
max    2016.000000        83.940002


*count:* O número total de observações no conjunto de dados. 3253 observações para o ano e a expectativa de vida.

*mean:* A média dos valores. Para o ano: 1908.07, e para a expectativa de vida: 48.68. 

*std:* O desvio padrão, que mede a dispersão dos valores em relação à média. Quanto maior o desvio padrão, mais dispersos os valores estão em relação à média. No caso do ano, o desvio padrão é aproximadamente 62.61, e para a expectativa de vida, é aproximadamente 17.97.

*min:* O valor mínimo observado. Para o ano: 1800 e para a expectativa de vida: 8.108836.

*25% (percentil 25):* O valor abaixo do qual 25% das observações estão. Por exemplo, para o ano, 25% das observações têm um valor menor ou igual a 1854.

*50% (percentil 50 ou mediana):* O valor que separa a metade inferior e a metade superior das observações. Para o ano, a mediana é 1908, o que significa que metade das observações são anteriores a 1908 e metade são posteriores a 1908.

*75% (percentil 75):* O valor abaixo do qual 75% das observações estão. Por exemplo, para a expectativa de vida, 75% das observações têm um valor menor ou igual a 66.82.

*max:* O valor máximo observado. Para o ano: 2016 e para a expectativa de vida: 83.94.

### contagem de valores nulos (NaN) em cada coluna do DataFrame:

In [12]:
print(data.isnull().sum())

Entity             0
Year               0
Life expectancy    0
dtype: int64


In [15]:
data.columns = data.columns.str.upper()
print(data.columns)

Index(['ENTITY', 'YEAR', 'LIFE EXPECTANCY'], dtype='object')


In [16]:
data.columns = data.columns.str.replace(" ", "_")
print(data.columns)

Index(['ENTITY', 'YEAR', 'LIFE_EXPECTANCY'], dtype='object')


### exclude year 1800 - 1989

In [18]:
filtered_data = data.query('YEAR >= 1990')

print(filtered_data.head())

        ENTITY  YEAR  LIFE_EXPECTANCY
188  Australia  1990        77.019997
189  Australia  1991        77.389999
190  Australia  1992        77.620003
191  Australia  1993        77.889999
192  Australia  1994        78.050003


In [ ]:
file_path = '/Users/emilytsen/Documents/projects/ds_cancer/df/life_expectancy.csv'

# Save the DataFrame to the CSV file
filtered_data.to_csv(file_path, index=False)